File: tests/fixtures/vulnerable_samples/python/sql_injection.py
"""Vulnerable Python code with SQL injection — for testing purposes ONLY.

This file intentionally contains security vulnerabilities for testing the Hunter.
Do NOT use this pattern in production code.
"""

import sqlite3
import sys

from flask import request


def get_user_by_name_vulnerable(db_path: str) -> list:
    """Vulnerable function: SQL injection via string concatenation.

    The user input from request.form is directly concatenated into the SQL query.
    Hunter should detect: source=request.form, sink=cursor.execute, CWE-89.
    """
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    # VULNERABLE: Direct string concatenation with user input
    user_input = request.form["username"]
    query = "SELECT * FROM users WHERE name='" + user_input + "'"
    cursor.execute(query)  # noqa: S608 — intentionally vulnerable for testing

    results = cursor.fetchall()
    conn.close()
    return results


def search_products_vulnerable(db_path: str) -> list:
    """Another SQL injection: using format string with user input."""
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    search_term = request.args["q"]
    # VULNERABLE: Format string with user input
    query = "SELECT * FROM products WHERE name LIKE '%" + search_term + "%'"
    cursor.execute(query)  # noqa: S608 — intentionally vulnerable for testing

    results = cursor.fetchall()
    conn.close()
    return results


def get_user_cli(db_path: str) -> list:
    """SQL injection from CLI arguments."""
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    user_id = sys.argv[1]
    # VULNERABLE: Direct CLI arg concatenation
    query = "SELECT * FROM users WHERE id=" + user_id
    cursor.execute(query)  # noqa: S608 — intentionally vulnerable for testing

    results = cursor.fetchall()
    conn.close()
    return results

File: tests/fixtures/vulnerable_samples/go/sql_injection.go
// Vulnerable Go code with SQL injection — for testing purposes ONLY.
// This file intentionally contains security vulnerabilities for testing the Hunter.
// Do NOT use this pattern in production code.

package main

import (
	"database/sql"
	"fmt"
	"net/http"
)

// GetUserVulnerable demonstrates SQL injection via string concatenation in db.Query.
// Hunter should detect: source=r.FormValue, sink=db.Query, CWE-89.
func GetUserVulnerable(db *sql.DB, w http.ResponseWriter, r *http.Request) {
	username := r.FormValue("username")

	// VULNERABLE: Direct string concatenation in SQL query
	query := "SELECT * FROM users WHERE name = '" + username + "'"
	rows, err := db.Query(query)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	defer rows.Close()
	fmt.Fprintf(w, "Query executed")
}

func main() {
	db, _ := sql.Open("sqlite3", ":memory:")
	http.HandleFunc("/user", func(w http.ResponseWriter, r *http.Request) {
		GetUserVulnerable(db, w, r)
	})
	http.ListenAndServe(":8080", nil)
}
